### PR TITLE
St/sdk 1434

### PR DESCRIPTION
**Pull request by saratnt**
## Description
This PR fixes that `debug_traceCall` RPC method incorrectly returns an error response for reverted transactions.
In addition, it adds a test for verifying that a transaction created with a gas limit = to its intrinsic gas is a valid transaction, even if it fails for out of gas.

## Jira Ticket
[SDK-1340](https://horizenlabs.atlassian.net/browse/SDK-1434?atlOrigin=eyJpIjoiMzhhMWU2NmY1OThiNGY5ZGFkODAzYmIzMWVhOTRhNjYiLCJwIjoiaiJ9)

## Changes
Changed exception handling in EthService.scala in `traceCall` method.

## Checks
- [x ] Project Builds
- [ x] Project passes tests and checks
- [ ] Updated documentation accordingly
- [ ] Breaking changes have been correctly tagged and notified
 
## Additional information
It needs the following pull requests:
- libevm https://github.com/HorizenOfficial/libevm/pull/19

[SDK-1340]: https://horizenlabs.atlassian.net/browse/SDK-1340?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by paologalligit — APPROVED** [review]

File: qa/sc_evm_native_interop.py [hunk 1]
```diff
@@ -157,6 +197,163 @@ def run_test(self):
...
+            {
+                "to": native_contract_address,
+                "input": data_input_failed
+            }, "latest"
+        )
+
+        assert_true('error' in result)
+        assert_true('write protection' in result['error']['message'])
+
+        # TODO traceCall seems to have problems with failed txs. Needs to check what is the correct behavior
```
Comment (alsala) [inline qa/sc_evm_native_interop.py:326]: the comment can be removed now that the traceCall have been fixed

**Review by alsala — APPROVED** [review]
Only one superminor comment

File: qa/sc_evm_native_interop.py (hunk 1 above)
Comment (saratnt) [reply qa/sc_evm_native_interop.py:326]: Done

**Comment by saratnt** [comment]
Merged into st/interop_final branch.

**Pull request closed without merging**